fl4health/model_bases/masked_layers/masked_normalization_layers.py

```diff
@@ -0,0 +1,297 @@
+            the weight and bias do not receive gradient in back propagation.
+            Instead, two score tensors - one for the weight and another for the bias - are maintained.
+            In the forward pass, the score tensors are transformed by the Sigmoid function into probability scores,
+            which are then used to produce binary masks via bernoulli sampling.
+            Finally, the binary masks are applied to the weight and the bias. During training,
+            gradients with respect to the score tensors are computed and used to update the score tensors.
+
+            When affine is False, _BatchNorm does not have weight or bias.
+            Under this condition, both score tensors are None and _MaskedBatchNorm acts in the same way as _BatchNorm.
+
+            Note: the scores are not assumed to be bounded between 0 and 1.
+
+        Args:
+            num_features: number of features or channels :math:`C` of the input
+            eps: a value added to the denominator for numerical stability.
+                Default: 1e-5
+            momentum: the value used for the running_mean and running_var
+                computation. Can be set to ``None`` for cumulative moving average
+                (i.e. simple average). Default: 0.1
+            affine: a boolean value that when set to ``True``, this module has
+                learnable affine parameters. Default: ``True``
+            track_running_stats: a boolean value that when set to ``True``, this
+                module tracks the running mean and variance, and when set to ``False``,
+                this module does not track such statistics, and initializes statistics
+                buffers :attr:`running_mean` and :attr:`running_var` as ``None``.
+                When these buffers are ``None``, this module always uses batch statistics.
+                in both training and eval modes. Default: ``True``
+
+            Attributes:
+                weight: the weights of the module. The values are initialized to 1.
+                bias:   the bias of the module. The values are initialized to 0.
+                weight_score: learnable scores for the weights. Has the same shape as weight. When applied
+                    to the default initial values of self.weight (i.e., all ones), this is equivalent to
+                    randomly dropping out certain features.
+                bias_score: learnable scores for the bias. Has the same shape as bias. When applied to
+                    the default initial values of self.bias (i.e., all zeros), it does not have any actual
+                    effect. Thus, bias_score only influences training when MaskedLayerNorm is created
+                    from some pretrained nn.LayerNorm module whose bias is not all zeros.
+        """
+        factory_kwargs = {"device": device, "dtype": dtype}
+        super().__init__(num_features, eps, momentum, affine, track_running_stats, **factory_kwargs)
```

Comment:
Genuine question, can we not simply pass `device=device, dtype=dtype` at the end of the `super().__init__`? I'm not sure, but that would mean we don't have to do through the creation of the dictionary rigmarole 🙂

Reply:
Yes we definitely can 😂